src/rgw/rgw_admin.cc

```diff
@@ -6232,7 +6232,7 @@ int main(int argc, const char **argv)
     RGWMetadataLog *meta_log = store->meta_mgr->get_log(period_id);
 
     formatter->open_array_section("entries");
-    for (; i < g_ceph_context->_conf->rgw_md_log_max_shards; i++) {
+    for (; i < g_ceph_context->_conf->get_val<int64_t>("rgw_md_log_max_shards"); i++) {
```

Comment:
I was going to make local variable for `rgw_md_log_max_shards` for instance.
`g_ceph_context->_conf->get_val<int64_t>("rgw_md_log_max_shards")` should not be called in same method several time.

Reply:
@shinobu-x sure, if you can find any instance;  I think we should hook up the caching api (and in core look at perfect hashing &c), but those can be separate changes;  it seems clear we need to move forward